166-course_upgrades

mbrancato31 opened the pull request:
- added checkmark, (it will look better when the creator buttons don't show)
- added new data requested by sponsors: (when creates, in the course and also when updating)
        - skill level
        - intended audience
        - pre requisite
-fix issue with enrolling (when a quiz is the first module, the student does not get enrolled)

mbrancato31 committed:
added capabilities

mbrancato31 committed:
Merge branch 'dev' into 166-course_upgrades

dmols reviewed (changes requested):
Only two issues I noticed: 

1.) When course is created without an image, the new fields (for difficulty, prereq, etc) are still undefined. 

This next thing is just something to be mindful of: 

Attempting to create a pdf module isn't possible because of the changes I made in Module_File_System. Idk how some changes carried over here and others didn't, but once the two merge to dev it should be fixed.

collyntheshots reviewed (changes requested):
As stated above the new info added seems to be set to undefined if no image is inserted
![Screenshot 2021-07-04 231105](https://user-images.githubusercontent.com/51330857/124412813-57195480-dd1d-11eb-8ef1-d593af60218e.png)

Further if you use the "update info" button and dont input anything into the new fields they dont have any date at all
![Screenshot 2021-07-04 231132](https://user-images.githubusercontent.com/51330857/124412860-731cf600-dd1d-11eb-9b28-8022fc951fdf.png)

mbrancato31 committed:
Merge branch 'dev' into 166-course_upgrades

mbrancato31 committed:
fixed issues, added grey checkmark

- fixed issue with creating course with no image
- fixed issue when editing course info
- added grey checkmark when module is not completed

Commit:
Fixed styling of categories textbox

dmols reviewed (approved):
Everything works great. Only suggestion is that I think for the case where module is unfinished, I noticed the RadioButtonUnchecked icon might look better than a grey checkmark. I was messing around with different icons and this one looks best for this case imo

Commit:
Deleted MyFiles page and sidebar link

Commit:
Moved LogOut button

mbrancato31 committed:
changed grey checked button to RadioButtonUncheckedIcon

mbrancato31 committed:
Merge branch 'dev' into 166-course_upgrades
